tests/test_checker.py

```diff
@@ -135,7 +136,10 @@ async def _test_update(
                 len(checker.get_external_data()),
                 expected_data_count,
             )
-            checker._checkers = [UpdateEverythingChecker]
+
+            if not critical_source_test:
+                checker._checkers = [UpdateEverythingChecker]
```

Comment:
There might be a cleaner way of handling this. The problem is if this line runs with the new tests, some failed assertions happen later. The new tests are still valid though as-is.

 I assume it's because some code here is missing handling for `x-checker-data`, (it only handles `extra-data`?).

Reply:
Fixed by moving to a new file